### PR TITLE
hack/build-go.sh: Use race flag when running go test for individual packages

### PR DESCRIPTION
**Pull request by jschintag**
**What this PR does / why we need it**:
Ensure that running tests indivdually is done in the same manner as when running all tests at once.

Currently running `make go-test` runs the tests with the `-race`. This is not the case when running tests individually with `make WHAT=pkg/some/package go-test`.

This PR adds the flag to test executed individually.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
Not needed, as it affects only how tests are run.
```

### CODE REVIEW

File: hack/build-go.sh [hunk 1]
```diff
@@ -103,7 +103,7 @@ fi
 for arg in $args; do
     if [ "${target}" = "test" ]; then
         (
-            go ${target} -v -tags "${KUBEVIRT_GO_BUILD_TAGS}" ./$arg/...
+            go ${target} -v -tags "${KUBEVIRT_GO_BUILD_TAGS}" - race ./$arg/...
```
Comment (alicefr) [inline hack/build-go.sh:106]: Is the emty space between `-` and `rage` correct?

**Comment by jschintag** [comment]
I mainly noted this discrepancy when testing current code with `make go-test` and having it fail.

The reason beieng that https://github.com/kubevirt/kubevirt/issues/8765 is not solved. Should i open a copy of that issue or can someone re-open it? It never got any attention and i'm not sure how important that problem is.

**Comment by alicefr** [comment]
> I mainly noted this discrepancy when testing current code with `make go-test` and having it fail.
> 
> The reason beieng that #8765 is not solved. Should i open a copy of that issue or can someone re-open it? It never got any attention and i'm not sure how important that problem is.

I would reopen it. We mainly use bazel and not go directly, that's probably the reason why we haven't noticed it

File: hack/build-go.sh [hunk 2]
```diff
@@ -103,7 +103,7 @@ fi
 for arg in $args; do
     if [ "${target}" = "test" ]; then
         (
-            go ${target} -v -tags "${KUBEVIRT_GO_BUILD_TAGS}" ./$arg/...
+            go ${target} -v -tags "${KUBEVIRT_GO_BUILD_TAGS}" -race ./$arg/...
```
Comment (alicefr) [inline hack/build-go.sh:106]: I only see `-race` when target isn't test: https://github.com/kubevirt/kubevirt/blob/main/hack/build-go.sh#L65

Comment (jschintag) [reply hack/build-go.sh:106]: `target` here is always test. When you run `make go-test` without any packages specified it currently runs the tests with `-race`. When you run `make WHAT=pkg/virt-launcher go-test` for example, it currently would not run with the flag. Thats why i added the flag here to ensure that the result remains the same.

Comment (alicefr) [reply hack/build-go.sh:106]: I see, thanks!

**Comment by alicefr** [comment]
Thanks for your contribution, let's be consistent and introduce the new flag
/approve

**Pull request merged**